cartridges/app_custom_account_verification/cartridge/controllers/Account.js

```diff
@@ -0,0 +1,187 @@
+            passwordConfirm: registrationForm.login.passwordconfirm.value,
+            validForm: registrationForm.valid,
+            form: registrationForm
+        };
+
+        if (registrationForm.valid) {
+            res.setViewData(registrationFormObj);
+
+            this.on('route:BeforeComplete', function (req, res) { // eslint-disable-line no-shadow
+
+                var authenticatedCustomer;
+                var serverError;
+
+                var registrationForm = res.getViewData(); // eslint-disable-line
+
+                if (registrationForm.validForm) {
+                    var newAccount;
+                    var login = registrationForm.email;
+                    var password = registrationForm.password;
+                    //save account data into custom object
+                    newAccount = accountVerifyHelpers.saveAccountToCO(registrationForm);
+                }
+
+                delete registrationForm.password;
+                delete registrationForm.passwordConfirm;
+                formErrors.removeFormValues(registrationForm.form);
+
+                if (serverError) {
+                    res.setStatusCode(500);
+                    res.json({
+                        success: false,
+                        errorMessage: Resource.msg('error.message.unable.to.create.account', 'login', null)
+                    });
+
+                    return;
+                }
+
+                if (registrationForm.validForm) {
+                    //send verification email
+                    accountVerifyHelpers.sendVerificationEmail(newAccount);
+
+                    res.setViewData({ authenticatedCustomer: authenticatedCustomer });
+                    res.json({
+                        success: true,
+                        redirectUrl: accountHelpers.getLoginRedirectURL(req.querystring.rurl, req.session.privacyCache, true)
+                    });
+
+                    req.session.privacyCache.set('args', null);
+                } else {
+                    res.json({
+                        fields: formErrors.getFormErrors(registrationForm)
+                    });
+                }
+            });
+        } else {
+            res.json({
+                fields: formErrors.getFormErrors(registrationForm)
+            });
+        }
+
+        return next();
+    }
+);
+
+/**
+ * Account-Verify : This endpoint is called when user click on the link of the sended verification email
+ * @name Base/Account-Verify
+ * @function
+ * @memberof Account
+ * @param {renders} - isml
+ * @param {serverfunction} - get
+ */
+server.get('Verify', function (req, res, next) {
+    var CustomObjectMgr = require("dw/object/CustomObjectMgr");
+    var URLUtils = require("dw/web/URLUtils");
+    var accountVerifyCO = 'ACCOUNT_VERIFY_CO'
+
+    var accountId = req.querystring.accountId;
+    var email = req.querystring.email;
+    var existingCustomer = CustomerMgr.getCustomerByLogin(email);
```

Comment:
Please use helper function for the logic below

Reply:
moved logic into helper function